PowerToys New+ variable support in template filenames -- documentation related updates (#5285)
MicrosoftDocs/windows-dev-docs: hub/powertoys/newplus.md

After the enablement toggle, the New+ Templates location setting is likely the most interesting one. By default, the template location is in your local app data folder, specifically at `%localappdata%\Microsoft\PowerToys\NewPlus\Templates`. However, these templates will not roam with you across devices. If you want a common set of templates across devices, a popular option is to change the template location to a folder that is synced with a cloud drive, such as OneDrive. This way, you can access your templates from any device.

> ### <a name="display_options"></a>Display options

Reviewer: What's the reason for adding the link with the underscore here? You can already link to this H3 with this URL: `https://learn.microsoft.com/en-us/windows/powertoys/newplus#display-options`

Author: I didn't realize that was possible, thank you for pointing that out!   I've removed the added bookmarks

The option enables you to toggle the display of starting digits, spaces and dots. When this option is toggled off (the default), a file named "1. filename" will be displayed as is. However, when this option is toggled on, the template will be displayed as "filename". This is useful when using digits, spaces, and dots at the beginning of filenames to control the display order of templates.

> ### <a name="behavior"></a>Behavior

Reviewer: Please remove these links. Headers are already linkable on the Learn site.

Author: I didn't realize that was possible, thank you for pointing that out!   I've removed the added bookmarks

#### <a name="replace_variables"></a>Replace variables in template filename

> With this option on (the default) supported variables in filenames, including in files within subfolders, will get replaced when the template is copied. 

Reviewer: ```suggestion
This setting causes supported variables in filenames, including in files within subfolders, to be replaced when the template is copied. The default setting of this option is enabled.
```

> Note: Any non-valid-filename characters are replaced with spaces.

Reviewer: ```suggestion
Note: Any invalid filename characters are replaced with spaces.
```

##### Examples

> | Example template filename | Would on copy expand to |

Reviewer 2: - Is it a "template filename" or a "file name template"?
- I think a phrasing like "would" sounds too insecure. "Result" is simple and confident.

Author: Filename:
We are using "filename" in the New+ settings UI and also in the existing New+ documentation. Both filename and file name are valid but started with filename and prefer to be consistent throughout.

Updated to "Result"

| :---             | :--- |
|`$YYYY-$MM-%DD, $hh $mm $ss  - $PARENT_FOLDER_NAME by %USERNAME%` | `2024-11-22, 12 08 54 - PowerShell project by cgaarden` |
|`File where variable value contains invalid characters %USERPROFILE%` | `File where variable value contains invalid characters C  Users cgaarden` |

##### Date and time related variables

> These date and time related variable patterns are the same as for PowerRename and are case-sensitive.

Reviewer: ```suggestion
These date and time related variable patterns are the same as those used by PowerRename and are case-sensitive.
```

| `$MM`            | Month, as digits with leading zeros for single-digit months. |
| `$M`             | Month, as digits without leading zeros for single-digit months. |
| `$DDDD`          | Name of the day of the week. |
| `$DDD`           | Abbreviated name of the day of the week. |
| `$DD`            | Day of the month, as digits with leading zeros for single-digit days. |
| `$D`             | Day of the month, as digits without leading zeros for single-digit days. |
| `$hh`            | Hours, with leading zeros for single-digit hours. |
| `$h`             | Hours, without leading zeros for single-digit hours. |
| `$mm`            | Minutes, with leading zeros for single-digit minutes. |
| `$m`             | Minutes, without leading zeros for single-digit minutes. |
| `$ss`            | Seconds, with leading zeros for single-digit seconds. |
| `$s`             | Seconds, without leading zeros for single-digit seconds. |
| `$fff`           | Milliseconds, represented by full three digits. |
| `$ff`            | Milliseconds, represented only by the first two digits. |
| `$f`             | Milliseconds, represented only by the first digit. |

##### Special variables

> These special variables are case-sensitive, so they will only work when written exactly as shown here.

Reviewer 2: Sentence can be shortened. Like, "Note: Case sensitive" should suffice.

Author: Not sure I agree - to me and you - yes :-) but I think there a lot of people who don't really know what case-sensitive/insensitive mean.

| Variable | Explanation |
| :---             | :--- |

> | `$PARENT_FOLDER_NAME`          | will expand to the name of the parent folder. |

Reviewer: ```suggestion
| `$PARENT_FOLDER_NAME`          | Expands to the name of the parent folder. |
```

##### Environment variables

> These variables are case-insensitive, meaning you can write them in a mix of uppercase or lowercase.

Reviewer 2: Same. Nit: to me, the word "write" sounds more like "handwriting", versus "type"

Author: Reworded slightly to hopefully avoid confusion

> Each `%environment_variable%` in the file and folder name, will get replaced by the value of that environment variable.

Reviewer: ```suggestion
Each `%environment_variable%` in the file and folder names will be replaced with the value of the corresponding environment variable.
```

> For instance, %USERNAME% will get replaced with the name of the current Windows user.

Reviewer: ```suggestion
For instance, %USERNAME% will be replaced with the name of the current Windows user.
```
